hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/upgrade/SevenToEightUpgradeHandler.java

```diff
@@ -62,19 +78,38 @@ public class SevenToEightUpgradeHandler implements UpgradeHandler {
   @Override
   public Map<ConfigProperty, String> upgrade(HoodieWriteConfig config, HoodieEngineContext context,
                                              String instantTime, SupportsUpgradeDowngrade upgradeDowngradeHelper) {
+    Map<ConfigProperty, String> tablePropsToAdd = new HashMap<>();
     HoodieTable table = upgradeDowngradeHelper.getTable(config, context);
     HoodieTableMetaClient metaClient = table.getMetaClient();
     HoodieTableConfig tableConfig = metaClient.getTableConfig();
+    // If auto upgrade is disabled, set initial version and writer version to 6 and return
+    if (!config.autoUpgrade()) {
+      setInitialVersion(config, table.getMetaClient().getTableConfig(), tablePropsToAdd);
```

Comment:
why do we need to set initial version in this case ? AFAIK, Initial version is not used to determine behavior

Reply:
It is not strictly needed. I just thought it would be nice to keep initial version consistent with writer table version.

Reply:
removed